content/blog/how_to_share_a_nextjs_app_from_localhost.md

```diff
@@ -0,0 +1,149 @@
+outputs:
+  - HTML
+  - AMP
+---
+
+{{< image "how_to_share_a_nextjs_app_from_localhost/banner.webp" "How to Share a Next.js App from Localhost" >}}
+
+Hosting your {{< link href="https://nextjs.org/" >}}Next.js{{< /link >}} app on the internet securely without deploying to a full server is easy with [Pinggy](https://pinggy.io). This guide will show you how to run your Next.js application locally and expose it publicly via a secure SSH tunnel in minutes.
+
+{{% tldr %}}
+1. **Run Next.js App**
+   - Create and set up your <a href="https://nextjs.org/" target="_blank">Next.js</a> app:
+     ```bash
+     npx create-next-app@latest my-app
+     cd my-app
+     npm run dev
+     ```
+2. **Create a Tunnel with Pinggy**
+   - Start SSH tunnel:
+     ```bash
+     ssh -p 443 -R0:localhost:3000 a.pinggy.io
+     ```
+   - Access your Next.js app via the provided [Pinggy](https://pinggy.io) public URL.
+{{% /tldr %}}
+
+{{< iframe src="https://www.youtube.com/embed/Di72BacDylw" title="YouTube video player" >}}
+
+## What Is Next.js?
+{{< link href="https://nextjs.org/" >}}Next.js{{< /link >}} is an open‑source React framework created by {{< link href="https://vercel.com/" >}}Vercel{{< /link >}} that streamlines building production‑ready web applications. Out of the box it provides:
+
+- **Static Site Generation (SSG)** for lightning‑fast, SEO‑friendly pages  
+- **Server‑Side Rendering (SSR)** to deliver dynamic content on request  
+- **Incremental Static Regeneration (ISR)** to update static content without redeploying  
+- **Built‑in Routing & API Routes** — no extra server configuration required  
+- **Automatic Code Splitting & Optimization** for smaller bundles and faster load times  
+- **Image & Font Optimization**, Internationalization (i18n), and Analytics integrations  
+
+Whether you’re prototyping a landing page, building a full e‑commerce storefront, or developing a complex dashboard, Next.js accelerates development while enforcing best practices.
+
+## What Is Pinggy?
```

Comment:
Remove "What Is Pinggy?" paragraph. Instead write that we can start next.js application in localhost, but others cannot access it because of CGNAT, nat etc. also describe that hosting it on a cloud is time consuming and expensive.